spinn_front_end_common/utilities/emergency_recovery.py

```diff
@@ -34,36 +34,23 @@ def _emergency_state_check():
             app_id, CPUState.RUN_TIME_EXCEPTION)
         watchdog_count = txrx.get_core_state_count(app_id, CPUState.WATCHDOG)
         if rte_count or watchdog_count:
-            states = txrx.get_cpu_infos(
-                None, [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG], True)
+            bad_cores = txrx.get_cores_in_states(
+                None, [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG])
             logger.warning(
                 "unexpected core states (rte={}, wdog={})",
                 rte_count, watchdog_count)
-            logger.warning(states.get_status_string())
+            logger.warning("cores {}", set(bad_cores))
     except Exception:
         logger.exception(
             "Could not read the status count - going to individual cores")
-        machine = txrx.get_machine_details()
-        infos = CPUInfos()
-        errors = list()
-        for chip in machine.chips:
-            for p in chip.processors:
-                try:
-                    info = txrx.get_cpu_information_from_core(
-                        chip.x, chip.y, p)
-                    if info.state in (
-                            CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG):
-                        infos.add_processor(chip.x, chip.y, p, info)
-                except Exception:
-                    errors.append((chip.x, chip.y, p))
-        if len(infos):
-            logger.warning(infos.get_status_string())
+        errors = txrx.get_cores_in_states(
+            CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG)
```

Comment:
This is a little more subtle - I think the idea was in the case of an error to go chip-by-chip to avoid getting a load of errors all at once

Reply:
Also note errors is when things go wrong in reading the state

Reply:
Yes the idea was to go chip by chip. BUT txrx.get_cpu_information_from_core did not support that.
Instead it gets all states and then returns the one wanted.